### PR TITLE
902 - "Launch" Slack alerts: Add documentation, surface menu item

### LINKED ISSUE #899: Slack integration: Surface in main menu
Surface the Slack integration section in the main menu, e.g. 
image

### LINKED ISSUE #902: Slack integration: Slack app setup documentation
Create the documentation for a user to set up a Slack integration for alerts.

Add it to this page: https://docs.growthbook.io/integrations/slack

Out of scope:

- Commands documentation

### PR DESCRIPTION
**Pull request by tinahollygb**
### Features and Changes

Adds documentation for configuring a Slack integration for alerts, and surfaces the menu item, essentially "launching" it.

**Note**: If we release this PR, some users won't see it until it's turned on for everyone. This may be confusing if they see it in the docs as this PR adds configuration instructions to the documentation as well.

See the docs: https://docs-81sqqv0lz-growthbook.vercel.app/integrations/slack

- Closes #902
- Closes #899 
- Closes https://linear.app/growthbook/issue/GB-34

### Dependencies

- [x] #919
- [x] #917 
- [x] #922 
- [x] https://github.com/growthbook/growthbook/pull/1034 (We want to add a feature flag)

This PR includes changes from the typed SDK PR [here](https://github.com/growthbook/growthbook/pull/1034). This PR's diff will shrink if we merge that one first, alternatively you can review both PR's in this PR.

### Testing

See the docs and app preview links.

### Screenshots

In the dashboard:

image

In the docs:

image

image

### CODE REVIEW

File: packages/front-end/components/Layout/Layout.tsx [hunk 1]
```diff
@@ -122,6 +122,22 @@ const navlinks: SidebarLinkProps[] = [
       },
     ],
   },
+  {
+    name: "Integrations",
+    href: "/integrations",
+    Icon: BsPlug,
+    path: /^(integrations)/,
+    autoClose: true,
+    feature: "slack-integration",
```
Comment (tinahollygb) [inline packages/front-end/components/Layout/Layout.tsx:131]: Only on for those where `slack-integration` is true. Right now this is configured off in production, on in dev, with a 50% rollout.

File: packages/front-end/components/Layout/Layout.tsx [hunk 2]
```diff
@@ -122,6 +122,22 @@ const navlinks: SidebarLinkProps[] = [
...
+    subLinks: [
+      {
+        name: "Slack",
+        href: "/integrations/slack",
+        path: /^integrations\/slack/,
+        feature: "slack-integration",
```
Comment (tinahollygb) [inline packages/front-end/components/Layout/Layout.tsx:137]: Also added support to conditionally turn on/off sub links. So when we're ready to release Slack integrations, we need to remove this from both the child and parent link. This support allows us to introduce new integrations without hiding the whole parent.

File: packages/front-end/components/Layout/SidebarLink.tsx [hunk 3]
```diff
@@ -127,61 +128,65 @@ const SidebarLink: FC<SidebarLinkProps> = (props) => {
             [styles.open]: open || selected,
           })}
         >
-          {props.subLinks.map((l) => {
-            if (l.superAdmin && !admin) return null;
+          {props.subLinks
+            .filter(
+              (subLink) => !subLink.feature || growthbook.isOn(subLink.feature)
+            )
```
Comment (tinahollygb) [inline packages/front-end/components/Layout/SidebarLink.tsx:134]: Adding this `.filter()` is the change—the rest is prettier auto-formatting the whitespace.

File: packages/front-end/components/Layout/SidebarLink.tsx [hunk 4]
```diff
@@ -128,61 +128,65 @@ const SidebarLink: FC<SidebarLinkProps> = (props) => {
             [styles.open]: open || selected,
           })}
         >
-          {props.subLinks.map((l) => {
-            if (l.superAdmin && !admin) return null;
+          {props.subLinks
+            .filter(
+              (subLink) => !subLink.feature || growthbook.isOn(subLink.feature)
```
Comment (bryce-fitzsimons) [inline packages/front-end/components/Layout/SidebarLink.tsx:133]: Should we also 404 the integrations/slack page itself if the feature is off? Personally I might not bother, just calling it out...

**Review by bryce-fitzsimons — APPROVED** [review]
LGTM, only minor feedback

File: packages/front-end/components/Layout/SidebarLink.tsx (hunk 4 above)
Comment (tinahollygb) [reply packages/front-end/components/Layout/SidebarLink.tsx:133]: I think it's ok if we leave the route alone. It's easier for us to test in prod if we wanted to before releasing the feature.

**Pull request merged**